File: PocketBeagle-2/rgb_led/color_circle/python/main.py
"""
This example cycles through all the base colors in the Color circle.
"""

from sysfs import Device
from pathlib import Path
from time import sleep

LED = Path("/sys/devices/platform/techlab-led/leds/multi-led/")
DELAY = 1

led = Device(path=LED)

max_brightness = led.sysfs("max_brightness").read_str()
# Set brightness to max from the start
led.sysfs("brightness").write(max_brightness)

multi_intensity = led.sysfs("multi_intensity")

while True:
    multi_intensity.write('255 0 0')
    sleep(DELAY)

    multi_intensity.write('255 255 0')
    sleep(DELAY)

    multi_intensity.write('0 255 0')
    sleep(DELAY)

    multi_intensity.write('0 255 255')
    sleep(DELAY)

    multi_intensity.write('0 0 255')
    sleep(DELAY)

    multi_intensity.write('255 0 255')
    sleep(DELAY)

    multi_intensity.write('255 255 255')
    sleep(DELAY)

    multi_intensity.write('0 0 0')
    sleep(DELAY)

File: PocketBeagle-2/rgb_led/color_circle/rust/src/main.rs
//! This example cycles through the base colors in the Color circle.

use std::{io::Write, thread::sleep, time::Duration};

use beagle_helper::sysfs::Device;

const LED: &str = "/sys/devices/platform/techlab-led/leds/multi-led/";
const DELAY: Duration = Duration::from_secs(1);

fn main() {
    let led = Device::with_path(LED).unwrap();
    let mut multi_intensity = led.sysfs_w("multi_intensity").unwrap();

    let max_brightness = led
        .sysfs_r("max_brightness")
        .unwrap()
        .read_string()
        .unwrap();
    // Set brightness to max from the start.
    led.sysfs_w("brightness")
        .unwrap()
        .write_all(max_brightness.as_bytes())
        .unwrap();

    loop {
        multi_intensity.write_all(b"255 0 0").unwrap();
        sleep(DELAY);

        multi_intensity.write_all(b"0 255 0").unwrap();
        sleep(DELAY);

        multi_intensity.write_all(b"0 0 255").unwrap();
        sleep(DELAY);
    }
}
